lib/syntax.ml

```diff
@@ -193,13 +212,24 @@ let rec resolve_columns env expr =
   each expr
 
 (** assign types to parameters where possible *)
-and assign_types { set_tyvar_strict; _ } expr =
+and assign_types env expr =
+  let { set_tyvar_strict; _ } = env in
   let option_split = function None -> None, None | Some (x,y) -> Some x, Some y in
   let rec typeof_ (e:res_expr) = (* FIXME simplify *)
     match e with
     | ResValue t -> e, `Ok t
     | ResParam p -> e, `Ok p.typ
     | ResInparam p -> e, `Ok p.typ
+    | ResInTupleList (param_id, kind) -> 
+      (match kind with 
+      | Res res_exprs -> ResInTupleList (param_id, ResTyped (List.map (fun expr ->
+         let typ = expr |> typeof |> snd |> get_or_failwith in 
+         if Type.is_any typ then 
+            fail "If you need to have a field as parameter in the left part you should specify a type"
```

Comment:
i believe everywhere else it is more lax - Any just gets casted to Text when it reaches code gen, lets keep it that way